version: '3.8'

services:
  app:
    build:
      context: ./app
      dockerfile: Dockerfile
    ports:
      - "8080:8080"
    volumes:
      - ./app:/app
      - /app/node_modules
    depends_on:
      - cdp
      - flask-app
    environment:
      - NODE_ENV=development
      - VITE_API_BASE_URL=http://localhost:5000

  cdp:
    build:
      context: ./cdp
      dockerfile: Dockerfile
    ports:
      - "3000:3000"
    volumes:
      - .:/app
      - ./cdp/wallet_seeds:/app/wallet_seeds
      - /app/node_modules
    environment:
      - NODE_ENV=developement

  flask-app:
    build:
      context: ./src
      dockerfile: Dockerfile
    ports:
      - "5000:5000"
    volumes:
      - ./src:/app
    depends_on:
      - db
    environment:
      - COINBASE_API_URL=http://cdp:3000
      - FLASK_ENV=development
      - FLASK_DEBUG=true
      - FLASK_APP=server.py
      - DB_NAME=chatbot
      - DB_USER=postgres  
      - DB_PASSWORD=postgres
      - DB_HOST=db
      - DB_PORT=5432
    command: flask run --host=0.0.0.0 --port=5000

  db:
    image: postgres:13
    environment:
      POSTGRES_DB: chatbot
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./src/initdb:/docker-entrypoint-initdb.d
    ports:
      - "5432:5432"

volumes:
  postgres_data:
